pkg/rpc/response/result/application_log.go

```diff
@@ -1,39 +1,41 @@
 package result
 
 import (
+	"encoding/json"
+	"errors"
+
 	"github.com/nspcc-dev/neo-go/pkg/core/state"
-	"github.com/nspcc-dev/neo-go/pkg/smartcontract"
 	"github.com/nspcc-dev/neo-go/pkg/util"
 	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
 )
 
 // ApplicationLog wrapper used for the representation of the
 // state.AppExecResult based on the specific tx on the RPC Server.
 type ApplicationLog struct {
-	TxHash      util.Uint256              `json:"txid"`
-	Trigger     string                    `json:"trigger"`
-	VMState     string                    `json:"vmstate"`
-	GasConsumed int64                     `json:"gasconsumed,string"`
-	Stack       []smartcontract.Parameter `json:"stack"`
-	Events      []NotificationEvent       `json:"notifications"`
+	TxHash      util.Uint256 `json:"txid"`
+	Trigger     string       `json:"trigger"`
+	VMState     string       `json:"vmstate"`
+	GasConsumed int64        `json:"gasconsumed,string"`
+	// Stack is either `[]stackitem.ItemJSON` or a string with marshaling error.
+	Stack  interface{}         `json:"stack"`
```

Comment:
These changes about stack are pretty good, but are you sure that they are valid? Because C# RPC client still has stack as array of *smartcontract parameters* (see [link](https://github.com/neo-project/neo-modules/blob/master/src/RpcClient/Models/RpcApplicationLog.cs#L19)). `stackitem.Item` and `smartcontract.Parameter` have slightly different serialisation to json: stackitem's ByteString -> smartcontract's ByteArray (types names are different), Struct -> Array, Pointer -> Any and etc. So I'm afraid that they can brake the compatibility.

Reply:
Oh, I confused invocation results with application logs. These changes are to be done for `result.Invoke`.
But it is funny, that for _marshaling_ `getapplicationlog` still uses raw stack items.